pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -161,40 +163,39 @@ func (c *Controller) updateOperatorStatus(ctx context.Context, previousStatus *o
 		errs = []error{}
 	}
 
-	deploymentAvailableCondition := operatorv1.OperatorCondition{
-		Type:   fmt.Sprintf("%sDeployment%s", c.conditionsPrefix, operatorv1.OperatorStatusTypeAvailable),
-		Status: operatorv1.ConditionTrue,
-	}
+	deploymentAvailableCondition := applyoperatorv1.OperatorCondition().
+		WithType(fmt.Sprintf("%sDeployment%s", c.conditionsPrefix, operatorv1.OperatorStatusTypeAvailable))
 
-	workloadDegradedCondition := operatorv1.OperatorCondition{
-		Type:   fmt.Sprintf("%sWorkloadDegraded", c.conditionsPrefix),
-		Status: operatorv1.ConditionFalse,
-	}
+	workloadDegradedCondition := applyoperatorv1.OperatorCondition().
+		WithType(fmt.Sprintf("%sWorkloadDegraded", c.conditionsPrefix))
 
-	deploymentDegradedCondition := operatorv1.OperatorCondition{
-		Type:   fmt.Sprintf("%sDeploymentDegraded", c.conditionsPrefix),
-		Status: operatorv1.ConditionFalse,
-	}
+	deploymentDegradedCondition := applyoperatorv1.OperatorCondition().
+		WithType(fmt.Sprintf("%sDeploymentDegraded", c.conditionsPrefix))
+
+	deploymentProgressingCondition := applyoperatorv1.OperatorCondition().
+		WithType(fmt.Sprintf("%sDeployment%s", c.conditionsPrefix, operatorv1.OperatorStatusTypeProgressing))
 
-	deploymentProgressingCondition := operatorv1.OperatorCondition{
-		Type:   fmt.Sprintf("%sDeployment%s", c.conditionsPrefix, operatorv1.OperatorStatusTypeProgressing),
-		Status: operatorv1.ConditionFalse,
+	status := applyoperatorv1.OperatorStatus()
+	if workload != nil {
+		status = status.WithGenerations(applyoperatorv1.GenerationStatus().
+			WithGroup("apps").
+			WithResource("deployments").
+			WithNamespace(workload.Namespace).
+			WithName(workload.Name).
+			WithLastGeneration(workload.Generation),
```

Comment:
not setting the generation will set it to `0`, right ?

Reply:
Yes, it should be set to the zero value